Utils/Extensions/StringExtensions.swift

```diff
@@ -52,6 +52,31 @@ public extension String {
         return str as! String
     }
 
+    /**
+    Truncates a String in the middle only if it's longer than `maxLength`
+
+      "ABCDEF".truncate(4)
+      // "AB…EF"
+
+    :param: maxLength The maximum length of the String.
+
+    :returns: A truncated String
+    */
+    func truncate(#maxLength: Int) -> String {
+        if count(self) >= maxLength {
+            let index1 = advance(self.startIndex, maxLength / 2)
```

Comment:
To fix the odd number problem, you need to do one of these things:

* Change this and line 72 to be `Int(ceil(Float(maxLength)…`.
* Fake it by changing `maxLength` on this line to `(maxLength + 1)`.

I recommend the latter, with a comment to note that this is essentially integer `ceil`.